src/App.tsx

```diff
@@ -15,47 +15,122 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  if (sortType === SortType.ALPHABET) {
+    visibleGoods.sort((a, b) => a.localeCompare(b));
+  }
+
+  if (sortType === SortType.LENGTH) {
+    visibleGoods.sort((a, b) => a.length - b.length);
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortGoodsByAlphabet = () => {
+    setSortType(SortType.ALPHABET);
+  };
+
+  const sortGoodsByLength = () => {
+    setSortType(SortType.LENGTH);
+  };
+
+  const reverseGoods = () => {
+    setIsReversed(!isReversed);
+  };
+
+  const resetGoods = () => {
+    setSortType(SortType.NONE);
+    setIsReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={
+            sortType === SortType.ALPHABET
+              ? 'button is-info'
+              : 'button is-info is-light'
+          }
+          onClick={sortGoodsByAlphabet}
         >
           Sort alphabetically
-        </button>
 
+        </button>
         <button
           type="button"
-          className="button is-success is-light"
+          className={
+            sortType === SortType.LENGTH
+              ? 'button is-info'
+              : 'button is-info is-light'
+          }
+          onClick={sortGoodsByLength}
         >
           Sort by length
-        </button>
 
+        </button>
         <button
           type="button"
-          className="button is-warning is-light"
+          className={
+            isReversed === true
```

Comment:
it's better to use classname library for making dynamic classNames 
```suggestion
            isReversed
```

Reply:
not fixed